packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -115,12 +120,26 @@ export interface AuthMiddleware {
  */
 const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
   const [params = {}] = args as [AuthMiddlewareParams?];
-  const { beforeAuth, afterAuth, publicRoutes, ignoredRoutes, apiRoutes, ...options } = params;
+  const publishableKey = params.publishableKey || PUBLISHABLE_KEY;
+  if (!publishableKey) {
+    errorThrower.throwMissingPublishableKeyError();
+  }
+  const secretKey = params.secretKey || SECRET_KEY;
+  if (!secretKey) {
+    errorThrower.throwMissingSecretKeyError();
+  }
```

Comment:
🔧  i think we shouldn't add the above 2 validations since they should be triggered by the `clerkClient.authenticateRequest()` of `@clerk/backend`. Having those checks makes them run for all our SDKs depending on the `@clerk/backend` and it makes it easier to maintain them being in a single place.

Reply:
I think these belong here as well. It's the library's responsibility to ensure that it's parameters are available as arguments or environment variables.

Reply:
From the offline discussion, we have concluded that those checks will be kept since the values of pk/sk are used outside of the `authenticateRequest()` context and it's confusing to depend that the authenticateRequest() should be called before using those values to validate the values.